### ISSUE TITLE
linter: `typescript-eslint/no-deprecated` incorrectly flags deprecated class members when `--type-aware` flag is set

### ISSUE THREAD

**Issue by AlCalzone** [oxc-project/oxc#15594]
### What version of Oxlint are you using?

1.28.0

### What command did you run?

```
oxlint --type-aware --deny=typescript-eslint/no-deprecated file.ts
```

### What does your `.oxlintrc.json` config file look like?

Does not need a config file to reproduce

### What happened?

Consider this code:
```ts
export class Test {
	/** @deprecated Use something else instead */
	public get foo(): number {
		return 42;
	}
}
```

This should not be an error if `Test.foo` is not being called. The above command outputs the following instead:
```
  × typescript-eslint(no-deprecated): `foo` is deprecated. Use something else instead
   ╭─[packages/zwave-js/src/test.ts:3:13]
 2 │     /** @deprecated Use something else instead */
 3 │     public get foo(): number {
   ·                ───
 4 │         return 42;
   ╰────

Found 0 warnings and 1 error.
```

If I remove the `--type-aware` flag, the error vanishes.

**Comment by camc314** [comment]
Thanks for reporting

**Comment by AlCalzone** [comment]
That was some fast turnaround time, thanks!
